x-pack/legacy/plugins/snapshot_restore/public/app/services/http/use_request.ts

```diff
@@ -90,37 +103,63 @@ export const useRequest = ({
       return;
     }
 
-    setError(response.error);
-    setData(response.data);
-    setLoading(false);
+    // Only set data if we are doing polling
```

Comment:
Nit: it looks like data is set in both conditions, so maybe this comment could be clarified.

Reply:
EDIT: I left another comment below which may make this moot.